### PR TITLE
[Multiple Datasource]set fixed totalcount number for empty state

### PR DESCRIPTION
**Pull request by yujin-emma**
### Description

Set the total count as 0 in empty state component instead of using extra props
### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6562

## Testing the changes

DataSourceAggregatedView - List active 
list-active

DataSourceAggregatedView - List all
list-all

DataSourceView
data-source-view

DataSourceSelectable
selectable

DataSourceMultiSelect
multi-select

## Changelog
- fix: Set fixed totalcount number for empty state

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by yujin-emma** [comment]
```
src/plugins/dashboard/public/application/components/dashboard_listing/dashboard_listing.test.tsx
Debugger listening on ws://127.0.0.1:53952/1c73741b-e44c-4e40-ac64-d42d9daa95b5
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
yarn run v1.22.19
$ scripts/use_node scripts/jest src/plugins/dashboard/public/application/components/dashboard_listing/dashboard_listing.test.tsx
Debugger listening on ws://127.0.0.1:53233/ec901f83-7176-4fe9-9533-162727657736
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
ciGroups []
jest-haste-map: duplicate manual mock found: index
  The following files share their name; please delete one of them:
    * <rootDir>/src/plugins/vis_type_vega/public/expressions/__mocks__/index.ts
    * <rootDir>/src/core/server/saved_objects/import/__mocks__/index.ts

 PASS  src/plugins/dashboard/public/application/components/dashboard_listing/dashboard_listing.test.tsx (5.455 s)
  dashboard listing
    ✓ renders table rows (141 ms)
    ✓ renders call to action when no dashboards exist (23 ms)
    ✓ hideWriteControls (47 ms)
    ✓ renders warning when listingLimit is exceeded (63 ms)
    ✓ render table listing with initial filters from URL (51 ms)

Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
Snapshots:   5 passed, 5 total
Time:        6.103 s
```
ciGroup3 failure is not related to code change in this PR

File: src/plugins/dashboard/public/application/components/dashboard_listing/__snapshots__/dashboard_listing.test.tsx.snap
Comment (zhongnansu) [inline src/plugins/dashboard/public/application/components/dashboard_listing/__snapshots__/dashboard_listing.test.tsx.snap:1]: is this change necessary?

File: src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx [hunk 1]
```diff
@@ -21,16 +21,10 @@ import { DataSourceDropDownHeader } from '../drop_down_header';
 import { DSM_APP_ID } from '../../plugin';
 
 interface DataSourceDropDownHeaderProps {
```
Comment (zhongnansu) [inline src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx:23]:
```suggestion
interface NoDataSourceProps {
```

File: src/plugins/dashboard/public/application/components/dashboard_listing/__snapshots__/dashboard_listing.test.tsx.snap
Comment (yujin-emma) [reply src/plugins/dashboard/public/application/components/dashboard_listing/__snapshots__/dashboard_listing.test.tsx.snap:1]: removed the unrelated change
